text/0102-offchain-parachain-runtime-upgrades.md

```diff
@@ -0,0 +1,354 @@
+# RFC-0000: Feature Name Here
+
+|                 |                                                                                             |
+| --------------- | ------------------------------------------------------------------------------------------- |
+| **Start Date**  | 13 July 2024                                                                                |
+| **Description** | Implement off-chain parachain runtime upgrades                                              |
+| **Authors**     | eskimor                                                                                     |
+
+## Summary
+
+Change the upgrade process of a parachain runtime upgrade to become an off-chain
+process with regards to the relay chain. Upgrades are still contained in
+parachain blocks, but will no longer need to end up in relay chain blocks nor in
+relay chain state.
```

Comment:
I mildly pushed for PVF upgrade to live in parablocks early on, but we descided for upgradfes on the relay chain since all validators need the data eventually anyways.  It's true however that (a) validator set churn makes off-chain an optimization, and being on-chain incurs extra costs, like repeated downloads.